packages/sdk-react/src/core/queries/useReceivables.ts

```diff
@@ -520,3 +521,81 @@ export function useInvoiceDetails({
     },
   };
 }
+
+export const useReceivableEmailPreview = (
+  receivable_id: string,
+  subject_text: string,
+  body_text: string
+): {
+  isLoading: boolean;
+  preview: string;
+  error: string;
+  refresh: () => void;
+} => {
+  const { i18n } = useLingui();
+  const { api } = useMoniteContext();
+
+  const mutation = api.receivables.postReceivablesIdPreview.useMutation(
+    {
+      path: {
+        receivable_id,
+      },
+    },
+    {
+      onError: (error) => {
+        const errorMessage = getAPIErrorMessage(i18n, error);
+        toast.error(
+          t(i18n)`Failed to update receivable line items: ${errorMessage}`
+        );
+      },
+    }
+  );
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [preview, setPreview] = useState<string>('');
+  const [error, setError] = useState<string>('');
+  const [attemptNumber, setAttemptNumber] = useState(0);
+
+  const refresh = () => {
+    setPreview('');
+    setError('');
+    setIsLoading(true);
+    setAttemptNumber(attemptNumber + 1);
+  };
+
+  const language = () => {
+    let locale = i18n.locale;
```

Comment:
In this project, we aim to avoid using ‘let’ and instead use ‘const’ for improved readability. The idea, as previously discussed with the team, is to prevent situations where ‘let’ is declared at the top of the file and then mutated further down. Therefore, it’s best to avoid using ‘let’ altogether in this case.

Reply:
addressed